doc/releases/migration-guide-3.6.rst

```diff
@@ -175,6 +175,9 @@ Device Drivers and Device Tree
                      <&rcc STM32_SRC_PLL1_Q FDCAN_SEL(1)>;
     };
 
+* Runtime configuration of UART driver is now by default disabled for Nordic UART driver. Motivation
+  for that change is that feature is rarely used and disabling reduces memory footprint.
```

Comment:
I'd rephrase this a bit. For example: "Runtime configuration is now disabled by default for Nordic UART drivers. The motivation for the change is that this feature is rarely used and disabling it significantly reduces the memory footprint."